### ISSUE TITLE
picky-asn1-x509 0.14 release has few build issues

### ISSUE THREAD

**Issue by elmarco** [Devolutions/picky-rs#325]
```
error: couldn't read `src/../../test_assets/missing_authority_key_identifier_field.crt`: No such file or directory (os error 2)
   --> src/certificate.rs:306:23
    |
306 |         let encoded = include_bytes!("../../test_assets/missing_authority_key_identifier_field.crt");
    |                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

And a couple of warnings:
```
warning: this function depends on never type fallback being `()`
   --> src/algorithm_identifier.rs:301:13
    |
301 | /             fn visit_seq<A>(self, mut seq: A) -> Result<Self::Value, A::Error>
302 | |             where
303 | |                 A: de::SeqAccess<'de>,
    | |______________________________________^
    |
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
    = note: for more information, see issue #123748 <https://github.com/rust-lang/rust/issues/123748>
    = help: specify the types explicitly
note: in edition 2024, the requirement `!: Deserialize<'de>` will fail
   --> src/macros.rs:14:14
    |
14  |         $seq.next_element()?.ok_or_else(|| {
    |              ^^^^^^^^^^^^
    |
   ::: src/algorithm_identifier.rs:359:25
    |
359 |                         seq_next_element!(seq, AlgorithmIdentifier, "sha algorithm identifier");
    |                         ----------------------------------------------------------------------- in this macro invocation
    = note: `#[warn(dependency_on_unit_never_type_fallback)]` on by default
    = note: this warning originates in the macro `seq_next_element` (in Nightly builds, run with -Z macro-backtrace for more info)

warning: creating a shared reference to mutable static is discouraged
   --> src/oids.rs:33:36
    |
33  |                   if let Some(oid) = &OID { oid.clone() } else { unreachable() }
    |                                      ^^^^ shared reference to mutable static
...
42  | / define_oid! {
43  | |     // x9-57
44  | |     DSA_WITH_SHA1 => dsa_with_sha1 => "1.2.840.10040.4.3",
45  | |     // x9-42
...   |
275 | |     LOCAL_PROTECTION_DESCRIPTOR => local_protection_descriptor => "1.3.6.1.4.1.311.74.1.8",
276 | | }
    | |_- in this macro invocation
    |
    = note: for more information, see issue #114447 <https://github.com/rust-lang/rust/issues/114447>
    = note: this will be a hard error in the 2024 edition
    = note: this shared reference has lifetime `'static`, but if the static ever gets mutated, or a mutable reference is created, then any further use of this shared reference is Undefined Behavior
    = note: `#[warn(static_mut_refs)]` on by default
    = note: this warning originates in the macro `define_oid` (in Nightly builds, run with -Z macro-backtrace for more info)
help: use `addr_of!` instead to create a raw pointer
    |
33  |                 if let Some(oid) = addr_of!(OID) { oid.clone() } else { unreachable() }
    |                                    ~~~~~~~~~   +

warning: creating a shared reference to mutable static is discouraged
   --> src/oids.rs:33:36
    |
33  |                   if let Some(oid) = &OID { oid.clone() } else { unreachable() }
    |                                      ^^^^ shared reference to mutable static
...
42  | / define_oid! {
43  | |     // x9-57
44  | |     DSA_WITH_SHA1 => dsa_with_sha1 => "1.2.840.10040.4.3",
45  | |     // x9-42
...   |
275 | |     LOCAL_PROTECTION_DESCRIPTOR => local_protection_descriptor => "1.3.6.1.4.1.311.74.1.8",
276 | | }
    | |_- in this macro invocation
    |
    = note: for more information, see issue #114447 <https://github.com/rust-lang/rust/issues/114447>
    = note: this will be a hard error in the 2024 edition
    = note: this shared reference has lifetime `'static`, but if the static ever gets mutated, or a mutable reference is created, then any further use of this shared reference is Undefined Behavior
    = note: this warning originates in the macro `define_oid` (in Nightly builds, run with -Z macro-backtrace for more info)
help: use `addr_of!` instead to create a raw pointer
    |
33  |                 if let Some(oid) = addr_of!(OID) { oid.clone() } else { unreachable() }
    |                                    ~~~~~~~~~   +

```

**Comment by CBenoit** [comment]
For the error which seems to be a build error when attempting to build the tests, I forward you to my answer here: https://github.com/Devolutions/picky-rs/issues/324#issuecomment-2496902875

We are relying on local ressources found in the git repository, but you are likely attempting to run the tests from the package without all the dev requirements. In this case, it seems harder to support properly: I think it’s a bad idea to package the test assets into the final package.

For the warnings, those are emitted by a newer version of rustc that we don’t use in our CI.
The version used is described in the rust-toolchain.toml file found here:
https://github.com/Devolutions/picky-rs/blob/f0105574876eb91040e3587c93c98fbce751ff92/rust-toolchain.toml

The toolchain should be updated and then the warnings fixed.

**Comment by elmarco** [comment]
@CBenoit what about having a common picky-test-data crate? that would be a good compromise size (not picked for normal build), ease of use etc?

**Comment by CBenoit** [comment]
Addressed by #327
